### PR TITLE
Check each field separately for imms which match everything

### PR DESCRIPTION
**Pull request by snuffysasa**
if they all pass make the whole line white,

If you approve this PR, then you can ignore/close the other, hot fix PR.

### CODE REVIEW

File: diff.py [hunk 1]
```diff
@@ -2441,8 +2438,28 @@ def do_diff(lines1: List[Line], lines2: List[Line], config: Config) -> Diff:
...
+                        if nf != of:
+                            # If the new field is a match to any symbol case
+                            # and the old field had a relocation, then ignore this mismatch
+                            if (
+                                field_matches_any_symbol(nf, config.arch)
+                                and line1.has_symbol
+                            ):
+                                continue
+                            all_fields_match = False
+
```
Comment (simonlindholm) [inline diff.py:2461]: This is a lot of code, let's put it in a function (i.e. reinstating imm_matches_everything but perhaps with a better name)

File: diff.py [hunk 2]
```diff
@@ -2441,8 +2438,28 @@ def do_diff(lines1: List[Line], lines2: List[Line], config: Config) -> Diff:
...
-                    if len(parts2) > 1 and imm_matches_everything(parts2[1], arch):
+
+                    all_fields_match = True
+                    new_parts = line2.scorable_line.split(None, 1)
+                    old_parts = line1.scorable_line.split(None, 1)
+                    newfields = new_parts[1].split(",")
+                    oldfields = old_parts[1].split(",")
+                    re_paren = re.compile(r"(?<!%hi)(?<!%lo)\(")
+                    oldfields = oldfields[:-1] + re_paren.split(oldfields[-1])
+                    newfields = newfields[:-1] + re_paren.split(newfields[-1])
```
Comment (simonlindholm) [inline diff.py:2449]: don't duplicate this code

File: diff.py [hunk 3]
```diff
@@ -2441,8 +2438,28 @@ def do_diff(lines1: List[Line], lines2: List[Line], config: Config) -> Diff:
...
+                            # If the new field is a match to any symbol case
+                            # and the old field had a relocation, then ignore this mismatch
+                            if (
+                                field_matches_any_symbol(nf, config.arch)
+                                and line1.has_symbol
+                            ):
+                                continue
+                            all_fields_match = False
+
+                    if all_fields_match:
```
Comment (simonlindholm) [inline diff.py:2462]: if there's different amounts of fields we also want to treat that as a mismatch

File: diff.py [hunk 4]
```diff
@@ -2222,7 +2218,8 @@ def diff_sameline(old_line: Line, new_line: Line) -> None:
         # Probably regalloc difference, or signed vs unsigned
 
         # Compare each field in order
-        newfields, oldfields = new.split(","), old.split(",")
+        new_parts, old_parts = new.split(None, 1), old.split(None, 1)
+        newfields, oldfields = new_parts[1].split(","), old_parts[1].split(",")
```
Comment (simonlindholm) [inline diff.py:2222]: can we add a helper function for "split line into mnemonic + arguments"?

File: diff.py (hunk 1 above)
Comment (snuffysasa) [reply diff.py:2461]: sounds good

File: diff.py (hunk 2 above)
Comment (snuffysasa) [reply diff.py:2449]: sounds good

File: diff.py (hunk 3 above)
Comment (snuffysasa) [reply diff.py:2462]: ahh, yep that's right, I meant to include that

File: diff.py (hunk 4 above)
Comment (snuffysasa) [reply diff.py:2222]: sure

**Pull request merged**